src/main/scala/li/cil/oc/util/FluidContainerUtils.scala

```diff
@@ -0,0 +1,211 @@
+package li.cil.oc.util
+
+import net.minecraft.inventory.IInventory
+import net.minecraft.item.ItemStack
+import net.minecraftforge.common.util.ForgeDirection
+import net.minecraftforge.fluids._
+
+object FluidContainerUtils {
+
+  /**
+   * Retrieves an actual fluid handler implementation for a fluid container item.
+   */
+  def fluidHandlerIn(inventory: IInventory, slot: Int): Option[IFluidHandler] = {
+    inventory.getStackInSlot(slot) match {
+      case stack: ItemStack if stack != null =>
+        val oneSizedStack = stack.copy()
+        oneSizedStack.stackSize = 1
+        oneSizedStack match {
+          case _ if FluidContainerRegistry.isFilledContainer(oneSizedStack) => Option(new FilledContainerWrapper(oneSizedStack))
+          case _ if FluidContainerRegistry.isEmptyContainer(oneSizedStack) => Option(new EmptyContainerWrapper(oneSizedStack))
+          case _ =>
+            stack.getItem match {
+              case _: IFluidContainerItem => Option(new FluidContainerItemWrapper(oneSizedStack))
+              case _ => None
+            }
+        }
+      case _ => None
+    }
+  }
+  
+  /**
+   * A fluid handler implementation that records fluid operations and can replay them.
+   */
+  def replayableFluidHandler(handler: IFluidHandler, simulate: Boolean = true): IFluidHandler = {
+    new ReplayableFluidHandler(handler, simulate)
```

Comment:
I don't quite get it, replayableFluidHandler is always called with simulate = true, and it is the only place that creates ReplayableFluidHandler, so it replays it's actions always in simulate mode. What am I missing?

Reply:
Ok it does not replay always in simulate mode, but the parameter still looks redundant

Reply:
Yes it is redundant, I want to keep this it can simulate or not the ability to simulate